packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -547,7 +553,20 @@ function insertNode(node: Node, parent: Node, anchor: Node | null, renderer: Ren
     if (process.env.NODE_ENV !== 'production') {
         unlockDomMutation();
     }
-    renderer.insert(node, parent, anchor);
+
+    // Handle vfragment-based nodes
+    if (leadingFragmentAnchors.has(node)) {
+        // The content of vfragment-based nodes are delimited by leading and trailing anchors.
+        const leading = node;
+        const content = renderer.nextSibling(node);
+        const trailing = renderer.nextSibling(content);
```

Comment:
I don't think it's safe to assume that the `nextSibling` of the `content` is the trailing text node. A fragment could have multiple `content` nodes, e.g.:

```html
<template lwc:If={foo}>
  <div>1</div>
  <div>2</div>
  <div>3</div>
</template>
```

Could you add a test for a fragment containing multiple child nodes like this? ^

Reply:
That's correct. We can't just assume that there's a single content node, so we can't use nextSibling. VFragments are [text, ...content, text] so there can be any number of children.